src/ddscxx/include/org/eclipse/cyclonedds/topic/datatopic.hpp

```diff
@@ -71,8 +72,26 @@ class ddscxx_serdata : public ddsi_serdata {
   bool& key_md5_hashed() { return m_key_md5_hashed; }
   const bool& key_md5_hashed() const { return m_key_md5_hashed; }
   void populate_hash();
-  T& getT() { return m_t; }
-  const T& getT() const { return m_t; }
+
+  T* getT()
+  {
+    if (!m_t)
+    {
+      m_t = new T();
```

Comment:
Good point. I amagined this would all be exectuted while holding some kind of lock. But if that is indeed not the case, then an atomic compare and swap like you suggest seems to be the way to go. I will look more into it and modify the code accordingly.

Reply:
Resolved the issue in line with @eboasson's proposal.